app/src/test/java/ca/klapstein/baudit/data/CareProviderTest.java

```diff
@@ -3,18 +3,136 @@
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
+import java.util.Arrays;
+import java.util.Collection;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+
+
+@RunWith(Parameterized.class)
 public class CareProviderTest {
-    @Before
-    public void setUp() {
+
```

Comment:
Alot is going on in this tests constructor making it difficult to pinpoint a failure if one does occur in it.

This should be re-thought and refactored. Or if at worst remove this test and implement it later.